### DISCUSSION TITLE
x265 3.6 release

### DISCUSSION THREAD

**Discussion by alatteri** [CESNET/UltraGrid#382] [General]
Hi UG people,

After a few years since the last x265 release, they cut 3.6.
https://bitbucket.org/multicoreware/x265_git/commits/tag/3.6

### Version 3.6
Release date - 4th April, 2024.
**New feature**
Segment based Ratecontrol (SBRC) feature
Motion-Compensated Spatio-Temporal Filtering
Scene-cut aware qp - BBAQ (Bidirectional Boundary Aware Quantization)
Histogram-Based Scene Change Detection
Film-Grain characteristics as a SEI message to support Film Grain Synthesis(FGS)
Add temporal layer implementation(Hierarchical B-frame implementation)
Enhancements to existing features
Added Dolby Vision 8.4 Profile Support

**API changes**
Add Segment based Ratecontrol(SBRC) feature: "--[no-]sbrc".
Add command line parameter for mcstf feature: "--[no-]mctf".
Add command line parameters for the scene cut aware qp feature: "--scenecut-aware-qp" and "--masking-strength".
Add command line parameters for Histogram-Based Scene Change Detection: "--hist-scenecut".
Add film grain characteristics as a SEI message to the bitstream: "--film-grain <filename>"
cli: add new option --cra-nal (Force nal type to CRA to all frames expect for the first frame, works only with keyint 1)
Optimizations
ARM64 NEON optimizations:- Several time-consuming C functions have been optimized for the targeted platform - aarch64. The overall performance increased by around 20%. SVE/SVE2 optimizations

**Bug fixes**
Linux bug to utilize all the cores
Crash with hist-scenecut build when source resolution is not multiple of minCuSize
32bit and 64bit builds generation for ARM
bugs in zonefile feature (Reflect Zonefile Parameters inside Lookahead, extra IDR issue, Avg I Slice QP value issue etc..)
Add x86 ASM implementation for subsampling luma
Fix for abrladder segfault with load reuse level 1
Reorder miniGOP based on temporal layer hierarchy and add support for more B frame
Add MacOS aarch64 build support
Fix boundary condition issue for Gaussian filter

**Comment by MartinPulec** [comment]
Is there some feature interesting for you? We currently use binary builds from [this PPA](https://launchpad.net/~savoury1/+archive/ubuntu/vlc3) in Linux, which should be 3.5. I hope/believe that this PPA will eventually upgrade to 3.6 at some time point. V3.6 should be already in other platform builds, however. I've just tested with one of [UVG video](https://ultravideo.fi/dataset.html) that the performance is almost the same as with v3.5 (using the x265 command line tool without any additional parameters).

**Reply by alatteri** [reply]
Not particularly, I just like to stay current and 3.5 was release quite awhile ago.  I did email Rob Savory regarding updating his PPA when 3.6 was release and he wanted to give it some more time to settle in.

**Reply by MartinPulec** [reply]
Thanks for the update and keeping it going - we'd perhaps keep the version from PPA for now - we could eventually build from sources as [some other FFmpeg deps](https://github.com/CESNET/UltraGrid/blob/master/.github/scripts/Linux/download_build_ffmpeg.sh), but there are not much advantages of doing that and hopefully it will eventually land into that PPA.

**Reply by MartinPulec** [reply]
The PPA has just updated x265 to 3.6 so it is already in current continuous Linux build. For the "stable" channel it will be used for next build (most likely 1.9.5).
